src/main/java/com/Elessar/database/MongoDB.java

```diff
@@ -0,0 +1,111 @@
+package com.Elessar.database;
+
+import com.Elessar.app.server.User;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import java.util.ArrayList;
+import java.util.List;
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.eq;
+
+/**
+ * Created by Hans on 1/20/19.
+ */
+public class MongoDB implements MyDatabase{
+    private final MongoDatabase db;
+    public MongoDB(MongoDatabase db) {
+        this.db = db;
+    }
+
+    @Override
+    public void insert(User user) {
+        final Document doc = new Document();
+        doc.append("name", user.getName());
+        doc.append("password", user.getPassword());
+        doc.append("email", user.getEmail());
+        doc.append("online", user.getOnline());
+        if (user.getPhoneNumber().length() > 0) {
+            doc.append("phone", user.getPhoneNumber());
+        }
+        db.getCollection("users").insertOne(doc);
+    }
+
+    @Override
+    public List<User> findUsers(User filter) {
+        List<User> findResult = new ArrayList<>();
+        try (MongoCursor<Document> cursor = db.getCollection("users").find(userToBson(filter)).iterator()) {
+            while (cursor.hasNext()) {
+                findResult.add(docToUser(cursor.next()));
+            }
+        }
+        return findResult;
+
+    }
+
+    @Override
+    public User update(User user) {
+        Bson filter = eq("name", user.getName());
+        if (user.getPassword() != null) {
+            filter = and(filter, eq("password", user.getPassword()));
+        }
+        Document prevUserDoc = db.getCollection("users").findOneAndUpdate(filter, new Document("$set", userToDoc(user)));
+        return docToUser(prevUserDoc);
+    }
+
+    private User docToUser(Document doc) {
+        if (doc == null) {
+            return null;
+        }
+        return new User(doc.getString("name"),
+                 doc.getString("password"),
+                 doc.getString("email"),
+                 doc.getString("phone"),
+                 doc.getString("online"));
+    }
+
+
+
+    private Bson userToBson(User filter) {
+        Bson bson = new BsonDocument();
```

Comment:
final

Reply:
bson value may get changed after this line, so I think this shouldn't use final

Reply:
I see, you are right